src/league.py

```diff
@@ -20,85 +20,94 @@ class League(object):
     def __init__(self, league_id: int, year: int, team_id: int, cookies, debug=False):
         self.league_id = league_id
         self.year = year
-        self.current_week = 0
-        self.nba_day = 0
         self.team_id = team_id
         self.cookies = cookies
-        if self.cookies:
-            self.cookies = cookies
-        else:
-            logger.error(f'No Authorization Credentials')
-            raise Exception
-
-        self._fetch_league()
-        self._fetch_team_id()
-        self._fetch_teams()
-        self.rosters = self._fetch_roster()
-        self.roster_stats = self._fetch_stats()
-        self.stat_totals = self._calculate_totals()
+        if not self.cookies:
+            logger.error('No Authorization Credentials')
+            raise ValueError('No Authorization Credentials')
 
     def __repr__(self):
-        return f'League: {self.league_id} Year: {self.year}'
-
-    def _fetch_league(self):
-        req = f'{FBA_ENDPOINT}{self.year}/segments/0/leagues/{self.league_id}'
-        resp = requests.get(req, params='', cookies=self.cookies)
-
-        self.status = resp.status_code
-        request_status(self.status)
+        return f'<League `{self.league_id}` {self.year}>'
 
-        league_data = resp.json()
-        logger.info(f'League Data: {json.dumps(league_data, indent=4)}')
+    def _fetch_league_meta(self):
 
-        self.current_week = league_data['status']['currentMatchupPeriod']
-        self.nba_day = league_data['status']['latestScoringPeriod']
+        url = f'{FBA_ENDPOINT}{self.year}/segments/0/leagues/{self.league_id}'
+        resp = requests.get(url=url, cookies=self.cookies)
 
-    def _fetch_team_id(self):
-        params = {
-            'view': 'mTeam'
-        }
+        # raise error if bad response
+        request_status(resp.status_code)
 
-        req = f'{FBA_ENDPOINT}{self.year}/segments/0/leagues/{self.league_id}'
-        resp = requests.get(req, params=params, cookies=self.cookies)
+        data = resp.json()
+        logger.info(f'League Data: {json.dumps(data, indent=4)}')
 
-        self.status = resp.status_code
-        request_status(self.status)
+        members = [(x['id'], x['displayName']) for x in data['members']]
+        teams = [(x['id'], x['owners'][0], x['nickname']) for x in data['teams']]
 
-        teams = resp.json()['teams']
+        return {
+            'current_week': data['status']['currentMatchupPeriod'],
+            'nba_day': data['status']['latestScoringPeriod'],
```

Comment:
Why remove 'current_week' and 'nba_day' as instance attributes?
having things like `self.current_week` and `self.nba_day` will make dashboard easier in long run- no?

Reply:
Yea... come to think of it, I agree.. If thats the case, can we add methods that are something like

`def __week___(self)` and in `def init(self): self.week = self.__week__()`.. we'd rather have that hidden...

Totally agree with you thought; my approach was to remove redundancy... it also screws up the mTeams or whatever call parameter.  I'll add that back

Reply:
+1 to that comment

Reply:
Discussed IRL, going to import meta at initialization